numpy/testing/_private/extbuild.py

```diff
@@ -0,0 +1,243 @@
+"""
```

Comment:
I took the opportunity to add this technique of compiling a c-extension module from fragments defined in a test, rather than having a seperate c-extension test module that we ship with NumPy. This keeps the code together with the test which is easier to maintain (PyPy uses this extensively). I did not yet build out the docstrings in the documentation.